src/classes/actions-cache/upload.ts

```diff
@@ -0,0 +1,180 @@
+import * as core from '@actions/core';
+import fs from 'fs';
+import {HttpClient} from '@actions/http-client';
+import {TypedResponse} from '@actions/http-client/lib/interfaces';
+import {ReserveCacheError, ValidationError} from '@actions/cache';
+import {isSuccessStatusCode} from './http-responses';
+import {retryHttpClientResponse, retryTypedResponse} from './retry';
+import {getOctokit} from '@actions/github';
+import {retry as octokitRetry} from '@octokit/plugin-retry';
+import {createHttpClient, getCacheApiUrl} from './http-client';
+
+const uploadFile = async (
+  httpClient: HttpClient,
```

Comment:
Again, too many parameters, let's stick to 3 max and when there's more, start representing them with object.

Reply:
well, the params does not seem to be related and their are unified through the module. There's no restrictions through the whole project so i prefer to leave it as is untill the whole repo refactoring with [eslint](https://eslint.org/docs/latest/rules/max-params) rule turned on.